macros/src/interface/generate.rs

```diff
@@ -262,20 +274,60 @@ fn generate_server(item_tokens: &mut TokenStream, fizyr_rpc: &syn::Ident, interf
 		}
 
 		/// An incoming message from a remote peer.
-		#visibility enum ReceivedMessage<#received_msg_generics>
+		#visibility enum ReceivedMessage<F>
 		where
-			#received_msg_where
+			F: #fizyr_rpc::util::format::Format,
 		{
 			#received_msg_variants
+
+			/// A stream message with an unknown service ID.
+			UnknownStream(#fizyr_rpc::Message<F::Body>),
+
+			/// A request with an unknow service ID.
+			UnknownRequest(#fizyr_rpc::ReceivedRequestHandle<F::Body>, F::Body),
+
+			/// A stream message with recognized service ID but invalid body.
+			///
+			/// The body has been consumed in the parse attempt,
```

Comment:
I understand, but is it maybe a bit unfortunate? I mean if we want to dump the body in case of parse errors, we now can't, right?

Reply:
Indeed. If you've got a suggestion I'd love to hear it. For messages without file descriptors it would be fine, you can't benefit from consuming the file descriptors. But for messages with file descriptors you can consume fds while parsing.